src/status.js

```diff
@@ -1,65 +1,96 @@
-          const li = document.createElement('li');
-          const text = `<div class="list-container"> <input class='check-input' type='checkbox' id='${i}' aria-label='...' checked>
-          <p class="todo-text">${task.itemText}</p><a class="del-menu" href="#"><i class="fas fa-ellipsis-v"></i></a></div>`;
-          li.classList.add('list-item');
-          li.innerHTML = text;
-          list.appendChild(li);
-        } else {
-          const li = document.createElement('li');
-          const text = `<div class="list-container"> <input class='check-input' type='checkbox' id='${i}' aria-label='...'>
-            <p class="todo-text">${task.itemText}</p><a class="del-menu" href="#"><i class="fas fa-ellipsis-v"></i></a></div>`;
-          li.classList.add('list-item');
-          li.innerHTML = text;
-          list.appendChild(li);
+
+      const todos = this.data.sort((a, b) => {
+        const indexA = a.todoIndex;
+        const indexB = b.todoIndex;
+
+        if (indexA < indexB) {
+          return -1;
         }
+        if (indexA > indexB) {
+          return 1;
+        }
+        return 0;
+      });
+      const li = document.createElement('li');
+      todos.forEach((task) => {
+        const text = `<div class="list-container main-list"> <input class='check-input' type='checkbox'  aria-label='...'>
+            <p class="todo-text">${task.todoText}</p><a class="del-menu" id="del" href="#"><i class="fas fa-ellipsis-v"></i></a></div>`;
+        li.classList.add('list-item');
+        li.innerHTML = text;
+        this.mainList.appendChild(li);
       });
-    }
+      localStorage.setItem('task', JSON.stringify(todos));
+      this.todoTask.value = '';
+      this.completeTodo();
+      this.removeData();
+    });
   }
 
-  getItems() {
-    return JSON.parse(localStorage.getItem('tasks'));
+  loadData() {
+    window.addEventListener('load', (e) => {
+      e.preventDefault();
+      const tasks = JSON.parse(localStorage.getItem('task'));
+      tasks.forEach((task) => {
+        const li = document.createElement('li');
+        const text = `<div class="list-container main-list"> <input class='check-input' type='checkbox'  aria-label='...'>
+              <p class="todo-text">${task.todoText}</p><a class="del-menu" id="del" href="#"><i class="fas fa-ellipsis-v"></i></a></div>`;
+        li.classList.add('list-item');
+        li.innerHTML = text;
+        this.mainList.appendChild(li);
+      });
+      this.completeTodo();
+      this.removeData();
+    });
   }
 
-  addItems(todo) {
-    localStorage.setItem('tasks', JSON.stringify(todo));
+  removeData() {
+    const del = document.querySelectorAll('#del');
+    del.forEach((v, i) => {
+      v.addEventListener('click', (e) => {
+        e.preventDefault();
+        const div = v.parentElement;
+        const li = div.parentElement;
+        li.remove();
+        const newTask = JSON.parse(localStorage.getItem('task'));
+        newTask.splice(i, 1);
```

Comment:
- [ ] Deleting a task should update all remaining items' indexes, so they represent the current list order and are unique. Currently, the function is not working as expected. After a click to remove the first element of the list, everything is wiped. Kindly double-check this and fix that issue